app/code/Magento/InventorySales/Api/ShippingAlgorithmResultInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Api;
+
+/**
+ * Responsible for how we will deduct product qty from different Sources
+ * @api
+ */
+interface ShippingAlgorithmResultInterface
```

Comment:
Move into `Inventory\Model\ShippingAlgorithm` folder
Supposed that will be extension point (SPI) of concrete implementation but not part of API